pinry-spa/src/main.js

```diff
@@ -1,17 +1,31 @@
 import Buefy from 'buefy';
 import Vue from 'vue';
 import { VueMasonryPlugin } from 'vue-masonry';
+import VueI18n from 'vue-i18n';
 import App from './App.vue';
 import router from './router';
 import setUpAxiosCsrfConfig from './components/utils/csrf';
 import './registerServiceWorker';
+import en from './assets/locales/en.json';
+
+const messages = {
+  en,
+};
 
 Vue.config.productionTip = false;
 Vue.use(Buefy);
 Vue.use(VueMasonryPlugin);
+Vue.use(VueI18n);
 setUpAxiosCsrfConfig();
 
+const i18n = new VueI18n({
+  locale: navigator.language.split('-')[0],
```

Comment:
While initializing the VueI18n, we'd better read something saved in local-storage or cookies to load user-prefered language more than auto-detection by navigator-language.

The priority is just like : `user-preference -> browser language -> fallback language`

Reply:
Hi @winkidney I agree we need to include user-preference local to local storage or cookies. I have added some code to address this requirement.

in the locale-changer I have added a onchange event that will set the locale into local storage.
```
            <div class="locale-changer">
              <select v-model="$i18n.locale" @change="setLocale($event)">
...
  methods: {
    setLocale(event) {
      localStorage.setItem('localeCode', event.target.value);
    },
...
```

once the above onchange event is triggered then the vue-i18n object will prioritize user preference in local storage, then browser language, then finally fallback local.

```
const i18n = new VueI18n({
  locale: localStorage.getItem('localeCode') || navigator.language.split('-')[0],
  fallbackLocale: 'en',
  messages,
});
```

Reply:
I have done some testing and since we have only one translations files it seems that we will hit only the browser language or fallback locale. When more translations files (ch.json, ja.json, es.json, etc...) is added then the locale will work as to the requirements previously specified.

Reply:
It's awesome, we may test it while we add another translation.
This feature would not be blocking thing of this awesome Pull Request.